src/Utils/kernel_launching.jl

```diff
@@ -43,10 +44,17 @@ function work_layout(grid, dims; include_boundaries=false)
                     (Int(√MAX_THREADS_PER_BLOCK), Int(√MAX_THREADS_PER_BLOCK))
 
     if include_boundaries
+        e = "include_boundaries kwarg requires the location kwarg to be specified"
+        isnothing(location) && throw(ArgumentError(e))
+
         TX, TY, TZ = topology(grid)
-        Nx′ = grid.Nx + (TX() isa Bounded)
-        Ny′ = grid.Ny + (TY() isa Bounded)
-        Nz′ = grid.Nz + (TZ() isa Bounded)
+        LX, LY, LZ = location
```

Comment:
can use `instantiate` from AbstractOperations (may want to move it to Utils)

Reply:
:+1: